### PR TITLE
Calculate value (#2317191)

### PR DESCRIPTION
**Pull request by penyaskito**

### CODE REVIEW

File: src/Plugin/Action/DataCalculateValue.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ *   },
+ *  provides = {
+ *     "result" = @ContextDefinition("float",
+ *       label = @Translation("Calculated result")
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add defined operation options from Drupal 7.
```
Comment (fago) [inline src/Plugin/Action/DataCalculateValue.php:41]: We need a todo also for better handling integers. Integers should be covered with the float action, but we need to ensure the result is an integer then also. As we cannot read the info of the actual context provided in any way yet, let's add a todo for handling integer context and results.

File: tests/src/Integration/Action/DataCalculateValueTest.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
```
Comment (fago) [inline tests/src/Integration/Action/DataCalculateValueTest.php:30]: Unnecessary new lines here.

File: tests/src/Integration/Action/DataCalculateValueTest.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    $this->action->setContextValue('input_1', $this->getTypedData('float', $input_1))
+      ->setContextValue('operator', $this->getTypedData('string', 'min'))
+      ->setContextValue('input_2', $this->getTypedData('float', $input_2));
+    $this->action->execute();
+    $result = $this->action->getProvided('result')->getContextValue();
+    $this->assertEquals(min($input_1, $input_2), $result, "Min calculation correct");
+  }
+
+  /**
+   * Tests the use of php's max function for 2 input values
```
Comment (fago) [inline tests/src/Integration/Action/DataCalculateValueTest.php]: Misses trailing point. Also we test the action not the php function.

File: tests/src/Integration/Action/DataCalculateValueTest.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    $this->action->setContextValue('input_1', $this->getTypedData('float', $input_1))
+      ->setContextValue('operator', $this->getTypedData('string', '/'))
+      ->setContextValue('input_2', $this->getTypedData('float', $input_2));
+    $this->action->execute();
+    $result = $this->action->getProvided('result')->getContextValue();
+    $this->assertEquals($input_1 / $input_2, $result, "Subtraction calculation correct");
+  }
+
+  /**
+   * Tests the use of php's min function for 2 input values
```
Comment (fago) [inline tests/src/Integration/Action/DataCalculateValueTest.php]: Misses trailing point. Also we test the action not the php function.

**Comment by penyaskito** [comment]
Updated with requested feedback. Didn't squash commits as I'm not sure if we want to lose track of changes, and not sure how github plays then with inline comments.

**Comment by klausi** [comment]
Continued in #153

**Pull request closed without merging**